fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/sorting/RecordQuerySortAdapter.java

```diff
@@ -71,10 +72,11 @@
     private Key encryptionKey;
     private static final Supplier<SecureRandom> RANDOM = Suppliers.memoize(SecureRandom::new);
 
-    protected RecordQuerySortAdapter(int memoryLimit, boolean memoryOnly,
+    protected RecordQuerySortAdapter(int memoryLimit, boolean memoryOnly, boolean isInsertionOrder,
                                      @Nonnull RecordQuerySortKey key, @Nonnull FDBRecordStoreBase<M> recordStore) {
         this.memoryLimit = memoryLimit;
         this.memoryOnly = memoryOnly;
+        this.isInsertionOrder = isInsertionOrder;
```

Comment:
Having this be a Boolean flag is a little awkward. It almost seems like this suggests that there needs to be more separation between the sorted cursor and the dam cursor, at the mechanics level, though I'm not sure I really believe that

Reply:
Yeah -- so if `isInsertionOrder` then we only use the comparator to find the current minimum key and treat everything that came before as smaller and everything that came afterwards as bigger. It makes the whole process stateful.

Reply:
I have removed this clunkiness in exchange for hopefully something better and less clunky.